File: Creating a simple plugin mechanism/[C#]-Creating a simple plugin mechanism/C#/SecondPlugin/SecondPlugin.cs
using PluginContracts;

namespace SecondPlugin
{
	public class SecondPlugin : IPlugin
	{
		#region IPlugin Members

		public string Name
		{
			get
			{
				return "Second Plugin";
			}
		}

		public void Do()
		{
			System.Windows.MessageBox.Show("Do Something in Second Plugin");
		}

		#endregion
	}
}

File: Creating a simple plugin mechanism/[VB]-Creating a simple plugin mechanism/VB/SecondPlugin/SecondPlugin.vb
Imports PluginContracts

Public Class SecondPlugin
    Implements IPlugin

    Public Sub DoSomething() Implements IPlugin.DoSomething
        System.Windows.MessageBox.Show("Do Something in Second Plugin")
    End Sub

    Public ReadOnly Property Name As String Implements IPlugin.Name
        Get
            Name = "Second Plugin"
        End Get
    End Property
End Class
